mono/eglib/glib.h

```diff
@@ -755,10 +751,7 @@ GUnicodeBreakType   g_unichar_break_type (gunichar c);
  * format must be a string literal, in order to be concatenated.
  * If this is too restrictive, g_error remains.
  */
-#define  g_assertf(x, format, ...) \
-	(G_LIKELY(x) ? 1 : (g_assertion_message ( \
-	"* Assertion at %s:%d, condition `%s' not met, function:%s, " format "\n", \
-	__FILE__, __LINE__, #x, __func__, __VA_ARGS__), 0))
+#define  g_assertf(x, format, ...) (G_LIKELY((x)) ? 1 : (g_assertion_message ("* Assertion at %s:%d, condition `%s' not met, function:%s, " format "\n", __FILE__, __LINE__, #x, __func__, __VA_ARGS__), 0))
```

Comment:
Double parens not needed on G_LIKELY, assuming it uses parens internally.

Reply:
I was expecting this change to put ## on VA_ARGS but I don't see it.

Reply:
@luhenry is addressing this here: https://github.com/mono/mono/pull/7458

Reply:
Indeed, extra parens don't fix it, but ## did with one compiler tested (clang).

$ cat 1.c

#define G_LIKELY(expr) (__builtin_expect ((expr) != 0, 1))

void g_assertion_message (const char *format, ...);

#define  g_assertf(x, format, ...) (G_LIKELY(x) ? 1 : (g_assertion_message ("* Assertion at %s:%d, condition `%s' not met, function:%s, " format "\n", __FILE__, __LINE__, #x, __func__, __VA_ARGS__), 0))

void f(int success)
{
g_assertf (success, "Failed to duplicate current thread handle");
}

$ gcc -c 1.c
1.c:11:1: error: expected expression
g_assertf (success, "Failed to duplicate current thread handle");
^
1.c:7:197: note: expanded from macro 'g_assertf'
  ...("* Assertion at %s:%d, condition `%s' not met, function:%s, " format "\n", __FILE__, __LINE__, #x, __func__, __VA_ARGS__), 0))
                                                                                                                              ^
1 error generated.

$ cat 1.c

#define G_LIKELY(expr) (__builtin_expect ((expr) != 0, 1))

void g_assertion_message (const char *format, ...);

#define  g_assertf(x, format, ...) (G_LIKELY(x) ? 1 : (g_assertion_message ("* Assertion at %s:%d, condition `%s' not met, function:%s, " format "\n", __FILE__, __LINE__, #x, __func__, ##__VA_ARGS__), 0))

void f(int success)
{
g_assertf (success, "Failed to duplicate current thread handle");
}

$ gcc -c 1.c
 => success

Reply:
Yes I saw that, that's why I expecting the ##. I was unaware of this problem otherwise.